app/controllers/concerns/mfa_setup_concern.rb

```diff
@@ -64,6 +64,15 @@ def in_account_creation_flow?
     user_session[:in_account_creation_flow] || false
   end
 
+  def mfa_selection_attempt_count
+    session[:mfa_attempts] ||= 0
+    session[:mfa_attempts] += 1
+  end
+
+  def reset_mfa_selection_attempt_count
+    session[:mfa_attempts] = nil
```

Comment:
We should probably delete the key altogether, rather than keeping it as a nil value.

```suggestion
    session.delete(:mfa_attempts)
```